Bad error message when using ansible module without ansible backend

Ansible module only work with ansible backend, but that's not clear:

```
>>> testinfra.get_host('local://').ansible.get_varialbes()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "testinfra/backend/base.py", line 247, in __getattr__
    module = getattr(self._host, testinfra.utils.un_camel_case(name))
  File "testinfra/host.py", line 89, in __getattr__
    assert name in testinfra.modules.modules, name + " is not a module"
AssertionError: get_variables is not a module
```

An access to `ansible` attribute should raise.

--- a/test/test_modules.py
+++ b/test/test_modules.py
@@ -276,11 +276,14 @@ def test_file(host):
 
 
 def test_ansible_unavailable(host):
+    expected = ('Ansible module is only available with '
+                'ansible connection backend')
     with pytest.raises(RuntimeError) as excinfo:
         host.ansible("setup")
-    assert (
-        'Ansible module is only available with ansible '
-        'connection backend') in str(excinfo.value)
+    assert expected in str(excinfo.value)
+    with pytest.raises(RuntimeError) as excinfo:
+        host.ansible.get_variables()
+    assert expected in str(excinfo.value)
 
 
 @pytest.mark.testinfra_hosts("ansible://debian_stretch")

--- a/testinfra/modules/ansible.py
+++ b/testinfra/modules/ansible.py
@@ -14,6 +14,7 @@
 from __future__ import absolute_import
 from __future__ import unicode_literals
 
+import functools
 import pprint
 
 from testinfra.modules.base import InstanceModule
@@ -35,6 +36,18 @@ class AnsibleException(Exception):
         self.result = result
         super(AnsibleException, self).__init__(
             "Unexpected error: {}".format(pprint.pformat(result)))
+
+
+def need_ansible(func):
+    @functools.wraps(func)
+    def wrapper(self, *args, **kwargs):
+        # pylint: disable=protected-access
+        if not self._host.backend.HAS_RUN_ANSIBLE:
+            raise RuntimeError((
+                "Ansible module is only available with ansible "
+                "connection backend"))
+        return func(self, *args, **kwargs)
+    return wrapper
 
 
 class Ansible(InstanceModule):
@@ -59,17 +72,15 @@ class Ansible(InstanceModule):
     """
     AnsibleException = AnsibleException
 
+    @need_ansible
     def __call__(self, module_name, module_args=None, check=True, **kwargs):
-        if not self._host.backend.HAS_RUN_ANSIBLE:
-            raise RuntimeError((
-                "Ansible module is only available with ansible "
-                "connection backend"))
         result = self._host.backend.run_ansible(
             module_name, module_args, check=check, **kwargs)
         if result.get("failed", False) is True:
             raise AnsibleException(result)
         return result
 
+    @need_ansible
     def get_variables(self):
         """Returns a dict of ansible variables
 

